gqcp/include/ONVBasis/SpinUnresolvedOperatorString.hpp

```diff
@@ -71,40 +73,72 @@ class SpinUnresolvedOperatorString {
+
+
     /*
      *  MARK: General information
      */
 
     /**
      *  Retrieve the operator indices from the `SpinUnresolvedOperatorString`.
+     * 
+     *  @return The vector containing the operator indices.
      */
     const std::vector<size_t>& operatorIndices() const { return this->indices; }
 
     /**
      *  Retrieve the phase factor corresponding to the `SpinUnresolvedOperatorString`.
+     * 
+     *  @return The phase factor associated with the operator string.
      */
     int phaseFactor() const { return this->p; }
 
     /**
      *  Check whether the operator string in question will result in zero when applied to the wave function.
      *
      *  Note: There are different cases when an operator string will result in a zero value. This method checks all of them.
+     * 
+     *  @return Whether the operator string in question will result in zero when applied to the wave function.
      */
-    bool isZero() const {
+    bool isZero() const;
 
-        // Check if the same index appears more than once in the operator string.
-        // Start by initializing a copy of the index vector associated with the operator string.
-        auto index_vector = this->operatorIndices();
 
-        // Since the operator string represents either only annihilation or creation operators, repetition of an index means that that index will be annihilated or created twice in any ONV following the operator string, which will automatically result in zero.
-        // We will use the std::unique function to check this condition. `std::unique` needs a sorted vector and removes all but the first instance of any unique group of elements. It then returns a sequence that's not necessarily equal to the original vector, as duplicate values are removed. Full explanation can be found here: https://stackoverflow.com/questions/46477764/check-stdvector-has-duplicates/46477901.
-        sort(index_vector.begin(), index_vector.end());
-        auto iterator = std::unique(index_vector.begin(), index_vector.end());
+    /*
+     *  MARK: Public methods
+     */
+
+    /**
+     *  Retrieve the phase factor after sorting the `SpinUnresolvedOperatorString`.
+     * 
+     *  Note: Please refer to method `sort()` to perform the actual sorting.
+     * 
+     *  @return The phase factor after sorting the operator string.
+     */
+    int phaseFactorAfterSorting();
 
-        // If nothing was removed by `std::unique`, i.e. all indices were unique, the return value of the iterator shall equal the last value of the vector.
-        // Since the operator string will not be zero in this case, we must return false.
-        return (iterator != index_vector.end());
-    }
+    /**
+     *  Sort the operator string (in-place) in ascending order and adjust its phase factor. If two indices are equal, the phase factor may not be correct but it does not really matter since its effect on a wavefunction is zero.
+     */
```

Comment:
Since you mention that the phase factor might be wrong if two indices are equal, wouldn't it be beneficial to first check whether or not the `operatorString` is zero? 

I think this can thus be somewhat circumvented by adding a `throw` statement if `isZero` returns True before sorting.